viewer/src/scripts/components/LeafletMap.jsx

```diff
@@ -40,47 +50,102 @@ var LeafletMap = React.createClass({
   map: null, 
   layer: null,
 
+  getInitialState: function () {
+    return {'value' : 0.5}
+  },
+
   componentDidMount: function () {    
     this.map = L.map(this.getDOMNode());
 
     baseLayers['Azavea'].addTo(this.map);    
     this.map.lc = L.control.layers(baseLayers).addTo(this.map);
     this.map.setView([30.25, -97.75], 4);
-
     this_map = this.map;
   },
 
-  render: function() {      
+  shouldComponentUpdate: function(nextProps, nextState) {
+    if(nextState.opacity) {  //if changing opacity
+      if(nextState.value !== this.state.value && this.layer) {
+        this.layer.setOpacity(nextState.value);
+      }
+      nextState.opacity = false;  //only allow handleOpacityUpdate to change opacity through Slider
+      return false;
+    } 
+    return true;
+  },
+
+  handleOpacityUpdate: function(event) {
+    this.setState({
+      value: event,
+      opacity: true
+    });
+  },
+
+  handleUpdateLayer: function() {
     var active = this.props.active; 
     var entry = this.props.active.entry;
-
-    if (this.isMounted() && active.entry) {      
-      this.map.setView([entry.center[1], entry.center[0]], entry.layer.zoom);
-      
+    this.map.setView([entry.center[1], entry.center[0]], entry.layer.zoom);
       var removeLayer = function (map, oldLayer){
         if (oldLayer) { 
           map.removeLayer(oldLayer) 
           map.lc.removeLayer(oldLayer);
         }
-
-
       }
-      
-      var oldLayer = this.layer;    
-      var args = active.band;      
-      args['breaks'] = active.entry.breaks.join(',');      
+     
+      var args = active.band; 
+      args['breaks'] = active.entry.breaks.join(',');
+
+      if(this.layer) {
+        var tmsUrl = this.layer._url
+        var findColorRegex = /&colorRamp=/;
+        tmsUrlSplit = tmsUrl.split(findColorRegex);
```

Comment:
this is a bit of a kludge and probably should be commented as such.  it's best to use a library to parse the URL and extract the parameter.  for example, this code breaks if the first query param was `colorRamp`:  `http://.../foo?colorRamp=`